plugins/triggerer.go

```diff
@@ -0,0 +1,209 @@
+package plugins
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/alexandre-normand/slackscot"
+	"github.com/alexandre-normand/slackscot/store"
+	"regexp"
+	"sort"
+	"strings"
+	"text/tabwriter"
+)
+
+// Triggerer holds the plugin data for the triggerer plugin
+// The triggerer plugin consists of a command to register a new trigger with its answer along
+// with a hear action that will listen and react if one of the registered triggers is said
+type Triggerer struct {
+	slackscot.Plugin
+	triggerStorer store.StringStorer
+}
+
+const (
+	// triggererPluginName holds identifying name for the triggerer plugin
+	triggererPluginName = "triggerer"
+)
+
+var registerTriggerRegex = regexp.MustCompile("(?i)trigger on (.+) with (.+)")
```

Comment:
@s-machina: I was considering changing the `regex` to surround the trigger/reaction with `"` but I'm not sure which one is best. It's kind of nice to have something more natural but the downside is that a reaction that would include `with` would mess things up.

Reply:
Maybe make quotes optional? Kind of like bash?

Reply:
That's an interesting idea. The only thing that annoys me with having this flexibility is that it complicates the `Usage` string.

Reply:
I'm going to go with this non-quoting version. The good news is that `trigger on deal with it with <url>` will still do the right thing. I'm including the version that will expect the `start-of-string` (`\A`) to fix the weird cases that would match both the register and delete regexes. 

We can tweak after gaining some experience from real world™ usage.